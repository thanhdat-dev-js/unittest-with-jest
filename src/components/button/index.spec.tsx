import { fireEvent, render } from "@test";
import renderer from "react-test-renderer";
import { Button } from "./index";

describe("Button component testing with testing-library", () => {
  it("renders without crashing", () => {
    const component = render(<Button onClick={() => undefined} />);

    expect(component).toBeTruthy();
  });
  it("button is clickable", () => {
    const mockFn = jest.fn();
    const { getByTestId } = render(<Button onClick={mockFn} />);

    const btn = getByTestId("btn");
    fireEvent.click(btn);
    fireEvent.click(btn);

    expect(mockFn).toHaveBeenCalledTimes(2);
  });
});

describe("Snapshot render", () => {
  it("should render correctly", () => {
    const tree = renderer
      .create(<a href="http://www.facebook.com">Facebook</a>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
