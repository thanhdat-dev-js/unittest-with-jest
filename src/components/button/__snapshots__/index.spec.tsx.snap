// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot render should render correctly 1`] = `
<a
  href="http://www.facebook.com"
>
  Facebook
</a>
`;
